com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/resolvers/v3/ModulePlaceholderResolver.java

```diff
@@ -0,0 +1,52 @@
+package com.sap.cloud.lm.sl.mta.resolvers.v3;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.sap.cloud.lm.sl.common.ContentException;
+import com.sap.cloud.lm.sl.mta.builders.v2.ParametersChainBuilder;
+import com.sap.cloud.lm.sl.mta.model.Hook;
+import com.sap.cloud.lm.sl.mta.model.Module;
+import com.sap.cloud.lm.sl.mta.model.RequiredDependency;
+import com.sap.cloud.lm.sl.mta.resolvers.PropertiesPlaceholderResolver;
+import com.sap.cloud.lm.sl.mta.resolvers.ResolverBuilder;
+
+public class ModulePlaceholderResolver extends com.sap.cloud.lm.sl.mta.resolvers.v2.ModulePlaceholderResolver {
+
+    public ModulePlaceholderResolver(Module module, String prefix, ParametersChainBuilder parametersChainBuilder,
+        ResolverBuilder propertiesResolverBuilder, ResolverBuilder parametersResolverBuilder, Map<String, String> singularToPluralMapping) {
+        super(module, prefix, parametersChainBuilder, propertiesResolverBuilder, parametersResolverBuilder, singularToPluralMapping);
+    }
+
+    @Override
+    public Module resolve() throws ContentException {
+        super.resolve();
+        module.setHooks(getResolvedHooks());
+        return module;
+    }
+
+    private List<Hook> getResolvedHooks() {
```

Comment:
Following the resolver patter, it would be a good idea to extract this to a HookResolver?